src/main/java/com/example/demo/model/Category.java

```diff
@@ -17,6 +18,7 @@
 @SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
 @Where(clause = "is_deleted=false")
 @Table(name = "categories")
+@RequiredArgsConstructor
```

Comment:
entity should have default constructor 
```suggestion
@NoArgsConstructor
```